File: 1089.js
const fs = require('fs');
const data = fs.readFileSync(0, 'utf8').trim().split(/\s+/).map(Number);

let proximo, anterior, count;
let idx = 0;

while (true){
    let n = data[idx++];

    if(n == 0) break;

    let v = [];
    for(let i = 0; i < n; i++){
        v.push(data[idx++]);
    }

    count = 0;
    for(let i = 0; i < n; i++){
        anterior = v[(i - 1 + n) % n];  
        proximo = v[(i + 1) % n];
        if((v[i] > anterior && v[i] > proximo) || (v[i] < anterior && v[i] < proximo)){
            count++;
        }
    }

    console.log(count);
}

File: 1089.py
while True:
    n = int(input())
    if n == 0:
        break

    v = list(map(int, input().split()))

    count = 0
    for i in range(n):
        anterior = v[(i - 1 + n) % n]
        proximo = v[(i + 1) % n]

        if (v[i] > anterior and v[i] > proximo) or (v[i] < anterior and v[i] < proximo):
            count += 1

    print(count)